docs/content/languages/java.md

```diff
@@ -0,0 +1,226 @@
+	URI url = req.getUrl();
+	String query = url.getQuery();
+	String[] pairs = query.split("&");
+	for (String pair : pairs) {
+		int idx = pair.indexOf("=");
+		query_pairs.put(URLDecoder.decode(pair.substring(0, idx), "UTF-8"), URLDecoder.decode(pair.substring(idx + 1), "UTF-8"));
+	}
+```
+
+#### Body 
+
+The body of the request object can be accessed as a map. You can use the map to convert to a value object using Jackson library's `ObjectMappper`. You will need to add the Jackson library to dependencies and import appropriate classes.
+
+```java
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+final ObjectMapper mapper = new ObjectMapper();
+HashMap data = (HashMap) req.getBody();
+Data iotData = mapper.convertValue(data, Data.class);
+```
+
+### Controlling HTTP Responses 
+
+#### Setting Response Headers & Status code
+
+The response object allows adding headers before sending the response back to the user. You can also set status code, body etc. on response object
+
+```java
+	HttpHeaders headers = new HttpHeaders();
+	headers.add("Access-Control-Allow-Origin", "*");
+	return ResponseEntity.status(HttpStatus.OK).headers(headers).build();
+```
+
+## Working with dependencies
+
+### Dependencies
+
+JVM environment can accept any executable JAR with entrypoint method that implements the interface of `io.fission.Function`. Currently the dependencies in the JVM environment are managed with Maven so we will take that as an example but you can use the others tools as well such as Gradle.
+
+JVM environment already has the spring-boot-starter-web and fission-java-core as dependencies so you need to declare them at provided scope. You can add additional dependencies that your application needs.
+
+```maven
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-web</artifactId>
+			<version>2.0.1.RELEASE</version>
+			<scope>provided</scope>
+		</dependency>
+		<dependency>
+			<groupId>io.fission</groupId>
+			<artifactId>fission-java-core</artifactId>
+			<version>0.0.2-SNAPSHOT</version>
+			<scope>provided</scope>
+		</dependency>
+```
+
+### Building the function
+
+The current build environment for Java has support for Maven builds. You can upload the source code and the JVM builder will build the source code into a jar. Let's take [Java example from here](https://github.com/fission/fission/tree/master/examples/jvm/java) and build using Fission builder.
+
+Let's first create a JVM environment with builder. For JVM environment you need to pass `--keeparchive` so that the jar file built from source is not extracted for running the function. You also need to use version 2 or higher of environment.
+
+```
+fission env create --name java --image fission/jvm-env --builder fission/jvm-builder --keeparchive --version 2
+```
+
+Next create a package with the builder environment by providing the source package.  This will kick off the build process.
+
+```
+$ zip java-src-pkg.zip -r *
```

Comment:
Better to show directory structure and command for arching. In case that users archive directory with the wrong path or structure.